Composer/packages/client/src/utils/hooks.ts

```diff
@@ -7,7 +7,11 @@ import { globalHistory } from '@reach/router';
 export const useLocation = () => {
   const { location, navigate } = globalHistory;
   const [state, setState] = useState({ location, navigate });
-  useEffect(() => globalHistory.listen(({ location }) => setState(state => ({ ...state, location }))), []);
+
+  useEffect(() => {
+    const removeListener = globalHistory.listen(({ location }) => setState(state => ({ ...state, location })));
+    return removeListener;
+  }, []);
```

Comment:
FWIW, I just wasn't sure of the return value of `globalHistory.listen`. I didn't mean to suggest a code change.